gnomad/utils/liftover.py

```diff
@@ -13,193 +13,148 @@
-        "Loading reference genomes, adding chain file, and loading fasta sequence for destination build"
-    )
-    if build == "GRCh38":
-        source = hl.get_reference("GRCh38")
+    logger.info("Loading fasta sequence for destination build...")
+    if input_build == "GRCh38":
         target = hl.get_reference("GRCh37")
-        chain = "gs://hail-common/references/grch38_to_grch37.over.chain.gz"
-        target.add_sequence(
-            "gs://hail-common/references/human_g1k_v37.fasta.gz",
-            "gs://hail-common/references/human_g1k_v37.fasta.fai",
-        )
+        chain = GRCH38_TO_GRCH37_CHAIN
+
     else:
-        source = hl.get_reference("GRCh37")
         target = hl.get_reference("GRCh38")
-        chain = "gs://hail-common/references/grch37_to_grch38.over.chain.gz"
-        target.add_sequence(
-            "gs://hail-common/references/Homo_sapiens_assembly38.fasta.gz",
-            "gs://hail-common/references/Homo_sapiens_assembly38.fasta.fai",
+        chain = GRCH37_to_GRCH38_CHAIN
+
+    logger.info("Adding liftover chain to input build...")
+    if source.has_liftover():
+        logger.warning(
+            f"Source reference build {source.name} already has a chain file!\
+            Using whichever chain has already been added."
+        )
+    else:
+        source.add_liftover(chain, target)
+
+    return (source, add_reference_sequence(target))
+
+
+def liftover_expr(
+    locus: hl.expr.LocusExpression,
+    alleles: hl.expr.ArrayExpression,
+    destination_reference: hl.ReferenceGenome,
+) -> hl.expr.StructExpression:
+    """
+    Generates struct liftover expression.
+
+    Struct contains:
+        - locus: Liftover coordinates
+        - alleles: Liftover alleles
+        - original_locus: Locus prior to liftover
+        - original_alleles: Alleles prior to liftover
+        - locus_fail_liftover: Whether the locus failed liftover
+        - ref_allele_mismatch: Whether the allele at index 0 of alleles (lifted over reference allele)
+            doesn't match the allele at that position in the destination reference
+
+    :param locus: Input locus.
+    :param alleles: Input alleles.
+    :param destination_reference: Desired reference genome build for liftover.
+    :return: Struct containing expressions for lifted over locus/alleles as well as original locus/alleles.
+    """
+    lifted_over_locus = hl.liftover(locus, destination_reference, include_strand=True)
+    lifted_over_alleles = alleles.map(
+        lambda a: hl.if_else(
+            lifted_over_locus.is_negative_strand, hl.reverse_complement(a), a
         )
+    )
 
-    source.add_liftover(chain, target)
-    return [source, target]
+    return hl.struct(
+        new_locus=lifted_over_locus.result,
+        new_alleles=lifted_over_alleles,
+        original_locus=locus,
+        original_alleles=alleles,
+        locus_fail_liftover=hl.is_missing(lifted_over_locus),
+        ref_allele_mismatch=lifted_over_locus.sequence_context()
```

Comment:
Does this need to be `ref_allele_mismatch=lifted_over_locus.result.sequence_context()` or  `ref_allele_mismatch=new_locus.sequence_context()`? Also similar to above, if `hl.liftover` fails to liftover, does this work on `None`?

Reply:
yes, good catch, thanks! this is missing a `.result`.